automation/src/test/java/com/microsoft/identity/common/test/automation/AcquireTokenPromptAutoGuestTest.java

```diff
@@ -0,0 +1,138 @@
+import net.thucydides.core.annotations.Managed;
+import net.thucydides.core.annotations.Steps;
+import net.thucydides.junit.annotations.TestData;
+
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.openqa.selenium.WebDriver;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collection;
+
+import io.appium.java_client.service.local.AppiumDriverLocalService;
+
+import static net.serenitybdd.screenplay.GivenWhenThen.seeThat;
+import static net.serenitybdd.screenplay.GivenWhenThen.then;
+import static org.hamcrest.Matchers.is;
+
+@RunWith(SerenityParameterizedRunner.class)
+public class AcquireTokenPromptAutoGuestTest {
+    @TestData
+    public static Collection<Object[]> FederationProviders() {
+
+        return Arrays.asList(new Object[][]{
+                {"ADFSv4"}
+        });
+    }
+
+    static AppiumDriverLocalService appiumService = null;
+
+    @Managed(driver = "Appium")
+    WebDriver hisMobileDevice;
+
+    @BeforeClass
+    public static void startAppiumServer() throws IOException {
+        appiumService = AppiumDriverLocalService.buildDefaultService();
+        appiumService.start();
+    }
+
+    @AfterClass
+    public static void stopAppiumServer() {
+        appiumService.stop();
+    }
+
+    private User james;
+    private String federationProvider;
+
+    @Steps
+    AcquireToken acquireToken;
+
+    @Steps
+    ReadCache readCache;
+
+    @Steps
+    ClickDone clickDone;
+
+    public AcquireTokenPromptAutoGuestTest(String federationProvider) {
+        this.federationProvider = federationProvider;
+    }
+
+    @Before
+    public void jamesCanUseAMobileDevice() {
+        TestConfigurationQuery query = new TestConfigurationQuery();
+        query.federationProvider = this.federationProvider;
+        query.isFederated = false;
+        query.userType = "Guest";
+        james = getUser(query);
+        james.can(BrowseTheWeb.with(hisMobileDevice));
+    }
+
+    private static User getUser(TestConfigurationQuery query) {
+
+        Scenario scenario = Scenario.GetScenario(query);
+
+        User newUser = User.named("james");
+        newUser.setFederationProvider(scenario.getTestConfiguration().getUsers().getFederationProvider());
+
+        // Because LAB API is returning wrong IDP, until this is fixed, the guest scenarios will be ran against ADFSv3
+        if(query.userType == "Guest"){
+            newUser.setFederationProvider("ADFSv3");
+        }
+        newUser.setTokenRequest(scenario.getTokenRequest());
+        newUser.setCredential(scenario.getCredential());
+        return newUser;
+    }
+
+
+    @Test
+    public void should_be_able_to_acquire_token() {
+
+        james.attemptsTo(
+                acquireToken.withPrompt("Auto"),
+                clickDone,
+                readCache);
+
+        then(james).should(seeThat(TokenCacheItemCount.displayed(), is(6)));
```

Comment:
This wouldn't work with latest common cache changes, the token cache count is 8 for common and 6 for Adal legacy. My PR here takes care of that  https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/1213/.
Once merged you can reference that and make the change